### DISCUSSION TITLE
Saving of consume and repeat settings

### DISCUSSION THREAD

**Discussion by WallyDW** [jcorporation/myMPD#1180] [Q&A] [answered]
I use the jukebox feature all the time, but it seems the settings I do is not saved over a restart.
It is consume, repeat and enable lyrics that seems to be losing its setting over a restart.
I have looked a bit in the logs and it seems consume require MPD 0.24, but I can not find anywhere to download such a version and enabling consume still seems to do the trick and get the jukebox feature going.
So what am I doing wrong here?

**Comment by jcorporation** [comment, marked as answer]
> I use the jukebox feature all the time, but it seems the settings I do is not saved over a restart.

Restart of what?

Some points:

- consume and repeat is not saved by myMPD. MPD saves this playback options in it's state file.
- consume command is available in all recent MPD versions, only the consume one-shot mode is introduced in MPD 0.24
- MPD 0.24 is not yet released
- the lyrics setting is saved and restored by myMPD.

**Comment by WallyDW** [comment]
Ohh, I restarted the entire system.
It never occurred to me that the consume and repeat was stored in MPD, but whenever I restart my MPD/myMPD sessions come up with playlist it closed with.
It then plays that list to finish and stop.
When I look at the settings, then repeat and consume is turned off again and there is a warning in the jukebox section about consume is required to be on.
I can manually turn the jukebox on again, which will set the consume and the repeat have to be set manually then.
I can also set repeat and consume manually and then turn the jukebox on manually again.

The lyrics part is working, so I might have confused myself with that one, probably missed a hit on the save button.

I have just confirmed that when I restart MPD, then I lose the repeat and consume setting, which then stops the jukebox.
Autoplay seems to be saved and keeps the setting though.
